tensordict/tensordict.py

```diff
@@ -144,18 +144,20 @@ def __init__(
         self.tensordict = tensordict
         self.include_nested = include_nested
         self.leaves_only = leaves_only
+        self.loop_set = set()
 
     def __iter__(self):
         return self._iter_helper(self.tensordict)
 
     def _iter_helper(self, tensordict, prefix=None):
         items_iter = self._items(tensordict)
-
+        self.loop_set.add(id(tensordict))
         for key, value in items_iter:
             full_key = self._combine_keys(prefix, key)
             if (
                 isinstance(value, (TensorDictBase, KeyedJaggedTensor))
                 and self.include_nested
+                and id(value) not in self.loop_set
```

Comment:
Is there a reason you opted for a set of ids here instead of setting some attribute on the TensorDict as in the case of `__repr__` and `flatten`?

Reply:
Yes, I preferred set because `_TensorDictKeysView` is independent class for keys retrieving.  What if we create several _TensorDictKeysView from same TensorDictBase, generally they couldn't work simultaneously.

Reply:
I see. I think it would be hard to use multiple key views simultaneously, but the set works just fine so let's leave it as it is 🙂